src/core/exception.d

```diff
@@ -47,29 +47,81 @@ else:
  */
 class RangeError : Error
 {
-    @safe pure nothrow this( string file = __FILE__, size_t line = __LINE__, Throwable next = null )
+    // lower/upper bound passed, if slice (array[lower .. upper]), if not a slice (array[upper])
+    size_t lower, upper;
+    //length of the indexed array
+    size_t length;
```

Comment:
I don't see a reason why these need to be mutable.

Reply:
I don't see any benefit of them being const or immutable, neither file, line nor message are.

Reply:
They probably should have been, but that's probably too late to change now. That doesn't justify that we cannot improve the other fields.

Reply:
It shows intent and will make debugging easier.